src/RecordCommand.cc

```diff
@@ -707,11 +711,37 @@ int RecordCommand::run(vector<string>& args) {
   if (running_under_rr()) {
     if (flags.nested == NESTED_IGNORE) {
       exec_child(args);
+    } else if (flags.nested == NESTED_DETACH) {
+      int ret = syscall(SYS_rrcall_detach_teleport, 0, 0, 0, 0, 0, 0);
+      if (ret < 0) {
+        FATAL() << "Failed to detach from parent rr";
+      }
+      if (running_under_rr(false)) {
+        FATAL() << "Detaching from parent rr did not work";
+      }
+      // Drop any LD_PRELOAD from the outer rr from our LD_PRELOAD list
+      char *ld_preload_env = getenv("LD_PRELOAD");
+      if (ld_preload_env) {
+        string ld_preload(ld_preload_env);
+        size_t it = ld_preload.find(":");
+        if (it != string::npos) {
+          // If the preload library is loaded at all, it must be first
+          size_t preload_it = ld_preload.find("librrpreload");
+          if (preload_it < it) {
+            string new_ld_preload=ld_preload.substr(++it);
+            setenv("LD_PRELOAD", new_ld_preload.c_str(), 1);
```

Comment:
`setenv` is a bit naff. Can we defer this processing until `RecordSession` sets up the environment of the tracee?

Reply:
Yeah, let me thread through the flag.